File: Hyper.java
import java.lang.Math;

public class Hyper {

   public static double cosh(double x) {
        return (Math.exp(x)+Math.exp(-x))/2.0;
    }
    public static double sinh(double x) {
        return (Math.exp(x)-Math.exp(-x))/2.0;
    }
    public static double tanh(double x) {
        return sinh(x)/cosh(x);
    } 
    public static double sech(double x) {
        return 1./cosh(x);
    }  
    public static double cotanh(double x) {
        return cosh(x)/sinh(x);
    }                    
}

File: Hyper.js
// Hyper.js
class Hyper {
    static cosh(x) {
        return (Math.exp(x) + Math.exp(-x)) / 2.0;
    }
    static sinh(x) {
        return (Math.exp(x) - Math.exp(-x)) / 2.0;
    }
    static tanh(x) {
        return this.sinh(x) / this.cosh(x);
    }
    static sech(x) {
        return 1.0 / this.cosh(x);
    }
    static cotanh(x) {
        return this.cosh(x) / this.sinh(x);
    }
}
